src/ZoneServer/Skills/Handlers/Clerics/Sadhu/Sadhu_Skill_Base.cs

```diff
@@ -0,0 +1,271 @@
+﻿using System;
+using System.Threading.Tasks;
+using Melia.Shared.Game.Const;
+using Melia.Shared.L10N;
+using Melia.Shared.World;
+using Melia.Zone.Network;
+using Melia.Zone.Skills.Combat;
+using Melia.Zone.World.Actors;
+using Melia.Zone.World.Actors.Characters;
+using Melia.Zone.World.Actors.Characters.Components;
+using Melia.Zone.World.Actors.CombatEntities.Components;
+using Melia.Zone.World.Items;
+using Yggdrasil.Logging;
+
+namespace Melia.Zone.Skills.Handlers.Clerics.Sadhu
+{
+	/// <summary>
+	/// Base skill class Sadhu skills.
+	/// </summary>
+	public class Sadhu_Skill_Base
+	{
+		/// <summary>
+		/// Handles a sadhu skill for the given BuffId
+		/// </summary>
+		/// <param name="skill"></param>
+		/// <param name="caster"></param>
+		/// <param name="originPos"></param>
+		/// <param name="farPos"></param>
+		/// <param name="target"></param>
+		/// <param name="buffId"></param>
+		public void Handle(Skill skill, ICombatEntity caster, Position originPos, Position farPos, ICombatEntity target, BuffId buffId)
```

Comment:
It's unfortunately not very clear what's going on here if you're not deep into this job already, made worse by all the duplicated code between `Handle` and `CreateClone`. Like with the buff handlers, I'd like to see a bit more structure that is easier to follow here.